js/stdlib/ts/CBR.ts

```diff
@@ -104,20 +104,24 @@ export const BT_Bytes = (len: number): BackendTy<CBR_Bytes> => ({
   name: `Bytes(${len})`,
   defaultValue: ''.padEnd(len, '\0'),
   canonicalize: (val: unknown): CBR_Bytes => {
-    const lenn = bigNumberToNumber(len);
-    if (typeof(val) !== 'string') {
-      throw Error(`Bytes expected string, but got ${j2s(val)}`);
-    }
-    const checkLen = (label:string, alen:number, fill:string): string => {
-      if ( val.length > alen ) {
-        throw Error(`Bytes(${len}) must be a ${label}string less than or equal to ${alen}, but given ${label}string of length ${val.length}`);
+    if (typeof(val) == 'string') {
+      const lenn = bigNumberToNumber(len);
+      const checkLen = (label:string, alen:number, fill:string): string => {
+        const v = val as string;
+        if ( v.length > alen ) {
+          throw Error(`Bytes(${len}) must be a ${label}string less than or equal to ${alen}, but given ${label}string of length ${v.length}`);
+        }
+        return v.padEnd(alen, fill);
+      };
+      if ( val.slice(0,2) === '0x' ) {
+        return checkLen('hex ', lenn*2+2, '0');
```

Comment:
This rule was always there, but we should add a note to the docs that this is a special case OR we should drop it because there's now the `Uint8Array` option